### PR TITLE
xapi: use more efficient protobuf structure for select result.

### PR DESCRIPTION
**Pull request by coocood**
Memory allocation reduced about 15%, performance improved about 3% on local store.

### CODE REVIEW

**Comment by coocood** [comment]
@BusyJay PTAL

**Comment by coocood** [comment]
@shenli @tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
LGTM

File: xapi/xapi.go [hunk 1]
```diff
@@ -180,6 +182,30 @@ func (pr *partialResult) Next() (handle int64, data []types.Datum, err error) {
...
+		}
+		rowMeta := chunk.RowsMeta[pr.cursor]
+		if !pr.ignoreData {
+			rowData := chunk.RowsData[pr.dataOffset : pr.dataOffset+rowMeta.Length]
+			data, err = tablecodec.DecodeValues(rowData, pr.fields, pr.index)
+			if err != nil {
+				return 0, nil, errors.Trace(err)
+			}
+		}
+		pr.dataOffset += rowMeta.Length
```
Comment (shenli) [inline xapi/xapi.go]: Move this to the above if.

**Comment by shenli** [comment]
LGTM

**Pull request merged**